src/main/java/io/kestra/plugin/youtube/VideoStats.java

```diff
@@ -0,0 +1,225 @@
+    title = "Get statistics for youtube videos",
+    description = "Retrieve detailed statistics for one more youtube videos including views, likes, comments"
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Get statistics for multiple videos",
+            full = true,
+            code = """
+                id: get_video_stats
+                namespace: company.team
+
+                tasks:
+                  - id: authenticate
+                    type: io.kestra.plugin.youtube.auth.OAuth2
+                    clientId: "{{ secret('YOUTUBE_CLIENT_ID') }}"
+                    clientSecret: "{{ secret('YOUTUBE_CLIENT_SECRET') }}"
+                    refreshToken: "{{ secret('YOUTUBE_REFRESH_TOKEN') }}"
+
+                  - id: get_stats
+                    type: io.kestra.plugin.youtube.task.VideoStats
+                    accessToken: "{{ outputs.authenticate.accessToken }}"
+                    videoIds:
+                      - "dQw4w9WgXcQ"
+                      - "9bZkp7q19f0"
+                    includeSnippet: true
+                """
+        )
+    }
+)
+public class VideoStats extends AbstractYoutubeTask implements RunnableTask<VideoStats.Output> {
+
+    @Schema(
+        title = "Video IDs",
+        description = "List of youtube video IDs to get statistics for"
+    )
+    @NotNull
+    private Property<List<String>> videoIds;
+
+    @Schema(
+        title = "Include snippet data",
+        description = "Whether to include video snippet data (title, description, thumbnail, etc.) in addition to statistics"
+    )
+    @Builder.Default
+    private Property<Boolean> includeSnippet = Property.ofValue(false);
+
+    @Schema(
+        title = "Include content details",
+        description = "Whether to include channels content details"
+    )
+    @Builder.Default
+    private Property<Boolean> includeContentDetails = Property.ofValue(false);
+
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        YouTube youTube = createYoutubeService(runContext);
+
+        List<String> renderedVideoIds = runContext.render(this.videoIds).asList(String.class);
+        boolean renderedIncludeSnippet = runContext.render(this.includeSnippet).as(Boolean.class).orElse(false);
+        boolean renderedIncludeContentDetails = runContext.render(this.includeContentDetails).as(Boolean.class).orElse(false);
+
+        // Build the parts parameter based on what data is requested
+        List<String> parts = new ArrayList<>();
+        parts.add("statistics");
+        if (renderedIncludeSnippet) {
+            parts.add("snippet");
+        }
+        if (renderedIncludeContentDetails) {
+            parts.add("contentDetails");
+        }
+
+        YouTube.Videos.List request = youTube.videos()
+            .list(Collections.singletonList(String.join(",", parts)))
+            .setId(Collections.singletonList(String.join(",", renderedVideoIds)))
+            .setMaxResults(50L);
```

Comment:
Maybe set this maxResults a property (with a control to avoid the user ask for a too big number)